tool/readme_scripts/readme_scripts_runner.dart

```diff
@@ -0,0 +1,184 @@
+    final filename = pathParts.last;
+    final lFilename = filename.toLowerCase();
+
+    // skip any files that aren't the readme or metadata files
+    if (lFilename != lReadme && lFilename != lMetadata) {
+      continue;
+    }
+
+    if (lFilename == lReadme) {
+      // exit if readme filename capitalization is incorrect
+      if (filename != 'README.md') {
+        print(
+            'Error: readme file has wrong capitalization in filename. Should be: README.md');
+        returnCode++;
+      }
+      // get the readme file
+      readmeFile = filename;
+    }
+
+    // get the metadata file
+    if (lFilename == lMetadata) {
+      // exit if metadata filename capitalization is incorrect
+      if (filename != 'README.metadata.json') {
+        print(
+            'Error: metadata file has wrong capitalization in filename. Should be: README.metadata.json');
+        returnCode++;
+      }
+      metadataFile = filePath;
+    }
+  }
+
+  // check the readme exists
+  if (readmeFile == null) {
+    print(
+        'Error: README.md does not exist. Please create and re-run the script.');
+    returnCode++;
+  }
+
+  // Run readme checks
+  // check filename capitalization
+  // run the markdown linter style checker on the readme
+  returnCode += runMdlStyleChecker(dirPath);
+
+  // run the readme content check
+  returnCode += runReadmeCheck(dirPath);
+
+  // if the metadata file doesn't exist, create it from the readme
+  if (metadataFile == null) {
+    returnCode += createMetadataFromReadme(dirPath, category);
+    // run the metadata content check
+    returnCode += runMetadataCheck(dirPath);
+  } else {
+    // run the metadata content check
+    returnCode += runMetadataCheck(dirPath);
+  }
+
+  if (returnCode != 0) {
+    exit(returnCode);
+  } else {
+    exit(0);
+  }
+}
+
+int createMetadataFromReadme(String dirPath, String category) {
+  print('**** Creating README.metadata.json from README.md ****');
+  var result = Process.runSync('python3',
+      ['create_metadata_from_README.py', '-s', dirPath, '-c', category]);
+  printScriptOutput('create readme from metadata script', result);
+  return result.exitCode;
+}
+
+List<String> getFilePaths(Directory directory) {
+  final files = <String>[];
+  final directoryContents = directory.listSync();
+  for (final file in directoryContents) {
+    if (file is File) {
+      files.add(file.path);
+    }
+  }
+  return files;
+}
+
+void printScriptOutput(String name, ProcessResult result) {
+  final stdout = result.stdout;
+  final stderr = result.stderr;
+  if (stdout != '') {
+    print(stdout);
+  }
+  if (stderr != '') {
+    print(stderr);
+  }
+  if (stdout == '' && stderr == '') {
+    print('$checkmark $name passed.');
+  }
+}
+
+int runMdlStyleChecker(String dirPath) {
+  print('**** Running mdl (markdown linter) style checker ****');
+  var result =
+      Process.runSync('mdl', ['--style', 'style.rb', '$dirPath/README.md']);
```

Comment:
Where does `mdl` come from? Should we spin up an issue to have `install_dependencies.sh` install `mdl` (and ruby and python)?

Reply:
Internal issue created.